flink-table/flink-sql-parser/src/test/java/org/apache/flink/sql/parser/FlinkSqlParserImplTest.java

```diff
@@ -1529,6 +1529,20 @@ public void testSetReset() {
         sql("RESET 'test-key'").ok("RESET 'test-key'");
     }
 
+    @Test
+    public void testTryCast() {
+        // Note that is expected that the unparsed value has the comma rather than AS, because we
+        // don't use a custom SqlNode for TryCast, but we rely on SqlBasicCall
+
+        // Simple types
+        expr("try_cast(a as timestamp)").ok("TRY_CAST(`A` AS TIMESTAMP)");
+        expr("try_cast('abc' as timestamp)").ok("TRY_CAST('abc' AS TIMESTAMP)");
+
+        // Complex types
+        expr("try_cast(a as row(f0 int, f1 varchar))")
```

Comment:
this is a type from Calcite, can we also try Flink complex types e.g.
`TRY_CAST(f0 AS ROW<a ARRAY<INT>, b MAP<STRING, DECIMAL(10, 2)>>)`

Reply:
Added this specific case